clang/lib/CodeGen/CGCall.cpp

```diff
@@ -5684,10 +5684,11 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
     if (!getLangOpts().FPAccuracyFuncMap.empty() ||
         !getLangOpts().FPAccuracyVal.empty()) {
       const auto *FD = dyn_cast_if_present<FunctionDecl>(TargetDecl);
-      assert(FD && "expecting a function");
-      CI = EmitFPBuiltinIndirectCall(IRFuncTy, IRCallArgs, CalleePtr, FD);
-      if (CI)
-        return RValue::get(CI);
+      if (FD) {
```

Comment:
When do we get a null FD here?   Can we assert that condition instead?

Reply:
We get the null when the entire sycl.hpp is included. And to use std::sin we need to include the entire sycl.hpp.

Reply:
I mean what property of this entity is it that causes FD to be NULL?   Can we assert (FD || <that condition>) instead of removing the non-NULL FD check that we had before?

Reply:
FD is null when TargetDecl is not a FunctionDecl. Not really sure what "_that condition_" should be?

Reply:
I chatted with @zahiraam offline; she believes that all the null FD cases are understood and handled here, so no assertion is necessary.

Reply:
Thanks @premanandrao for the updates about NULL cases.

Reply:
Just an additional comment about the assert.
For any TargetDecl, the control flow passes through this code. In the case we are using the fp-accuracy we need to check that the TargetDecl is a FunctionDecl and only then we can generate a call to the intrinsic. For any other kind of TargetDecl we can just create the call (in this particular case, the TargetDecl was a CXXConstructorDecl).